File: src/main/java/com/netcracker/skillstable/model/ApiResponse.java
package com.netcracker.skillstable.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.springframework.http.HttpStatus;

@Getter
@Setter
public class ApiResponse<T> {
    private int httpStatusCode;
    private HttpStatus status;
    private String message;
    private T result;

    @Builder
    public ApiResponse(HttpStatus status, String message, T result) {
        this.httpStatusCode = status.value();
        this.status = status;
        this.message = message;
        this.result = result;
    }
}

File: src/main/frontend/skills-table/src/app/model/api.response.ts
export class ApiResponse {
  httpStatusCode: number;
  status: string;
  message: string;
  result: any;

  constructor(httpStatusCode: number, status: string, message: string, result: any) {
    this.httpStatusCode = httpStatusCode;
    this.status = status;
    this.message = message;
    this.result = result;
  }
}
